src/screens/Connectors/ConnectorMetaData/BankDebit/BankDebit.res

```diff
@@ -0,0 +1,178 @@
+      name: "string",
+      onBlur: _ => (),
+      onChange: ev => {
+        let value = ev->Identity.formReactEventToString
+        let getPaymentMethodsObject = connector => {
+          open BankDebitTypes
+          {
+            payment_method: paymentMethod,
+            payment_method_type: paymentMethodType,
+            connector_name: connector,
+            mca_id: getConnectorId(
+              connector->ConnectorUtils.getConnectorNameTypeFromString(
+                ~connectorType=PMAuthenticationProcessor,
+              ),
+            ),
+          }
+        }
+        if value->LogicUtils.isNonEmptyString {
+          let paymentMethodsObject = value->getPaymentMethodsObject
+          setCurrentSelection(_ => value)
+
+          let existingPaymentMethodsArray =
+            enabledList.value->getArrayDataFromJson(BankDebitUtils.itemToObjMapper)
+
+          let newPaymentMethodsArray =
+            existingPaymentMethodsArray->Array.filter(item =>
+              item.payment_method_type !== paymentMethodType
+            )
+
+          newPaymentMethodsArray->Array.push(paymentMethodsObject)
+          enabledList.onChange(newPaymentMethodsArray->Identity.anyTypeToReactEvent)
+        }
+      },
+      onFocus: _ => (),
+      value: currentSelection->JSON.Encode.string,
+      checked: true,
+    }
+    <SelectBox.BaseDropdown
+      allowMultiSelect=false
+      buttonText="Select Value"
+      input
+      options
+      hideMultiSelectButtons=false
+      showSelectionAsChips=true
+      customButtonStyle="w-full"
+      fullLength=true
+      dropdownClassName={`${options->PaymentMethodConfigUtils.dropdownClassName}`}
+    />
+  }
+}
+
+@react.component
+let make = (
+  ~onCloseClickCustomFun,
+  ~setShowWalletConfigurationModal,
+  ~update,
+  ~paymentMethod,
+  ~paymentMethodType,
+  ~setInitialValues,
+) => {
+  open LogicUtils
+  open BankDebitUtils
+  let connectorList = HyperswitchAtom.connectorListAtom->Recoil.useRecoilValueFromAtom
+  let formState: ReactFinalForm.formState = ReactFinalForm.useFormState(
+    ReactFinalForm.useFormSubscription(["values"])->Nullable.make,
+  )
+  let form = ReactFinalForm.useForm()
+
+  let connectorsListPMAuth =
+    connectorList->ConnectorUtils.getProcessorsListFromJson(
+      ~removeFromList=ConnectorTypes.PMAuthenticationProcessor,
+    )
+  let pmAuthConnectors =
+    connectorsListPMAuth->Array.map(item => item.connector_name)->removeDuplicate
+
+  let pmAuthConnectorOptions = pmAuthConnectors->dropdownOptions
+
+  let getConnectorId = (connector: ConnectorTypes.connectorTypes) => {
+    let connectorData = connectorsListPMAuth->Array.find(item => {
+      item.connector_name->ConnectorUtils.getConnectorNameTypeFromString(
+        ~connectorType=ConnectorTypes.PMAuthenticationProcessor,
+      ) === connector
+    })
+    switch connectorData {
+    | Some(connectorData) => connectorData.merchant_connector_id
+    | None => ""
+    }
```

Comment:
Please confirm that this part is working as equality doesn't work with `variants with constructor` .

Reply:
updated